### ISSUE TITLE
Binary builds fail to build post v12.1 

### ISSUE THREAD

**Issue by pnasrat** [openshift/openshift-azure#2120]
Not sure if this is working as intendend but with `master` but appears to be a regression

Steps to reproduce

1. Create and login to a cluster using `make create` and copy login command and run in terminal
2. `oc new-project load-test`
3. `oc new-build --strategy docker --binary --docker-image fedora:31 --name pnasrat-load`

Create a Dockerfile that uses `FROM fedora:31` 

```
$ oc start-build pnasrat-load --from-dir . --follow
Uploading directory "." as binary input for the build ...
.
Uploading finished
Error from server (InternalError): Internal error occurred: pods "pnasrat-load-1-build" is forbidden: cannot exec into or attach to a privileged container
```

https://docs.openshift.com/container-platform/3.11/dev_guide/dev_tutorials/binary_builds.html#binary-builds-private-code

**Comment by pnasrat** [comment]
Created a second cluster using pluginconfig.yaml to override back to v12.1 and confirm was working 

```
oc start-build pnasrat-load --from-dir . --follow
Uploading directory "." as binary input for the build ...
..
Uploading finished
build.build.openshift.io/pnasrat-load-1 started
Receiving source from STDIN as archive ...
Replaced Dockerfile FROM image fedora:31
Pulling image fedora@sha256:8fa60b88e2a7eac8460b9c0104b877f1aa0cea7fbc03c701b7e545dacccfb433 ...
Pulled 1/1 layers, 100% complete
Extracting
Step 1/8 : FROM fedora@sha256:8fa60b88e2a7eac8460b9c0104b877f1aa0cea7fbc03c701b7e545dacccfb433
 ---> f0858ad3febd
...
```

**Comment by pnasrat** [comment]
The error message and cancellation appears to be from an admission controller - I'm guessing `vendor/k8s.io/kubernetes/plugin/pkg/admission/exec/admission.go` from the error string but will look more tomorrow

**Comment by olga-mir** [comment]
can it be related to privileged container work that went in v13?
image needs to be whitelisted in order to be allowed: https://github.com/openshift/openshift-azure/blob/9a4a503c0af4bd3edcc1c39dfbaf099ecb04e4eb/pkg/startup/v13/data/master/etc/aro-admission-controller/aro-admission-controller.yaml

**Comment by Makdaam** [comment]
SO after spending some time on it: this is caused by the DenyEscalatingExec, because start-build tries to exec/attach to the build container. I'm trying to find a way to make it work.

**Comment by ehashman** [comment]
Fixed in #2170.

**Comment by ehashman** [comment]
Released in v15.0, closing.
